os/binfmt/elf.c

```diff
@@ -300,6 +299,19 @@ static int elf_loadbinary(FAR struct binary_s *binp)
 	binp->alloc[1] = loadinfo.ctoralloc;
 	binp->alloc[2] = loadinfo.dtoralloc;
 #endif
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	binp->alloc[3] = (FAR void *)loadinfo.roalloc;
+	binp->alloc[4] = (FAR void *)loadinfo.dataalloc;
+#endif
+#endif
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	binp->textsize = loadinfo.textsize;
+	binp->rosize = loadinfo.rosize;
+	binp->datasize = loadinfo.datasize;
```

Comment:
When we don't do optimization of reloading time, are they necessary? I mean that why they are under CONFIG_APP_BINARY_SEPARATION.

Reply:
Yes. They are not required. I corrected it.